src/variety/pencils.js

```diff
@@ -0,0 +1,804 @@
+			if(cell.lcnt !== 1) {continue;}
+			
+			var cells = cell.getPencilCells();
+			
+			if(cells.length >= cell.path.nodes.length - 1) {continue;}
+
+			this.failcode.add("pencilSmallLength");
+			if(this.checkOnly){ return;}
+			cell.seterr(1);
+			cells.seterr(1);
+			cell.path.setedgeerr(1);
+		}
+	},
+
+	pencilNumberTooSmall: function() {
+		for(var c=0;c<this.board.cell.length;c++){
+			var cell = this.board.cell[c];
+			var dir = cell.qdir || cell.anum;
+			if(!(dir >= 1 && dir <= 4)) {continue;}
+			if(cell.lcnt !== 1) {continue;}
+			
+			var cells = cell.getPencilCells(cell.path.nodes.length - 1);
+			
+			if(cells.some(function(cellb){return cellb.qnum > 0 && cellb.qnum < cell.path.nodes.length - 1;})) {
+				this.failcode.add("bkSizeGt");
+				if(this.checkOnly){ return;}
+				cells.seterr(1);
+			}
+		}
+	},
+
+	unusedCells: function() {
+		var empty = this.board.cell.filter(function(c) { return c.lcnt === 0; });
+
+		for(var c=0;c<this.board.cell.length;c++){
+			var cell = this.board.cell[c];
+			var dir = cell.qdir || cell.anum;
+			if(!(dir >= 1 && dir <= 4)) {continue;}
+			if(cell.lcnt !== 1) {continue;}
+
+			var cells = cell.getPencilCells(cell.path.nodes.length - 1);
+			cells.each(function(c) {empty.remove(c);});
+		}
+
+		if(empty.length > 0) {
+			this.failcode.add("unusedCell");
+			if(this.checkOnly){ return;}
+			empty.seterr(1);
+		}
+	},
+
+	pencilExactAreas : function() {
+		for(var c=0;c<this.board.cell.length;c++){
+			var cell = this.board.cell[c];
+			var dir = cell.qdir || cell.anum;
+			if(!(dir >= 1 && dir <= 4)) {continue;}
+			if(cell.lcnt !== 1) {continue;}
+
+			var cellb = this.board.emptycell;
+			if(dir === cell.UP) { cellb = cell.adjacent.bottom; }
+			if(dir === cell.DN) { cellb = cell.adjacent.top; }
+			if(dir === cell.LT) { cellb = cell.adjacent.right; }
+			if(dir === cell.RT) { cellb = cell.adjacent.left; }
+
+			var area = cellb.room.clist.getRectSize();
+			if(area.cols * area.rows !== cell.path.nodes.length - 1 || (area.cols !== 1 && area.rows !== 1)) {
+				this.failcode.add("pencilExactArea");
+				if(this.checkOnly){ return;}
+				cell.seterr(1);
+			}
+		}
+	},
+
+	unusedBorders: function() {
+		var list = new this.klass.CellList();
+
+		for(var c=0;c<this.board.cell.length;c++){
+			var cell = this.board.cell[c];
+			var dir = cell.qdir || cell.anum;
+			if(!(dir >= 1 && dir <= 4)) {continue;}
+			if(cell.lcnt !== 1) {continue;}
+
+			var cells = cell.getPencilCells(cell.path.nodes.length - 1);
+			list.extend(cells);
+		}
+
+		var bds = this.board.border;
+		
+		for(var b=0; b<bds.length; b++) {
+			var border = bds[b];
+			if(border.qans === 1 && !list.include(border.sidecell[0]) && !list.include(border.sidecell[1])) {
+				this.failcode.add("unusedBorder");
+				if(this.checkOnly){ return; }
+				border.seterr(1);
+			}
+		}
+	}
+},
+
+FailCode:{
+	bkSizeLt : ["<number too large>","A number is bigger than the size of the pencil."],
```

Comment:
my approach to gathering Japanese translations has been to release in the form

    "(please translate) A number is bigger than the size of the pencil.";

I'd suggest doing the same here. (And I seem to remember the angled brackets causing issues because they end up in the HTML directly...

Reply:
Done